Pagemacro keyboard shortcuts

hamishwillee opened the pull request:
As per #3196 we are removing the Page macro (transclusion).

Prior to this change, [All keyboard shortcuts](https://developer.mozilla.org/en-US/docs/Tools/Keyboard_shortcuts) contains all shortcuts, parts of which are transcluded into other docs - e.g. [Debugger Keyboard shortcuts](https://developer.mozilla.org/en-US/docs/Tools/Debugger/Keyboard_shortcuts) and [Eyedropper](https://developer.mozilla.org/en-US/docs/Tools/Eyedropper)

Where possible, this PR **moves** the content from the  [All keyboard shortcuts](https://developer.mozilla.org/en-US/docs/Tools/Keyboard_shortcuts) and then links from the old page to the new page. 

I went back and forth a few times considering perhaps:
- just duplicating the content in both place
- Just linking from the location where the macro was to the sll keyboard shortcut docs. In the end I decided that while it is nice to have a single page with all the keyboard shortcuts, it is more useful for end users to have that information on hand when they are using the tool. 
 
I am not certain this was the best decision, mostly because now the shortcuts are scattered. I do think that that if you're using web console, page inspector etc you probably are happy enough with the current way. If you think I'd be better off duplicating or linking to the All ... doc instead let me know - relatively easy fix.

hamishwillee committed:
Make footnotes a bit more readable and fix note

hamishwillee commented on files/en-us/tools/style_editor/index.html:
```diff
@@ -107,4 +107,7 @@ <h2 id="Keyboard_shortcuts">Keyboard shortcuts</h2>
 
-<p>{{ Page ("en-US/docs/tools/Keyboard_shortcuts", "source-editor") }}</p>
+
+<p>See <a href="/en-US/docs/Tools/Keyboard_shortcuts#source_editor">All keyboard shortcuts > Source editor</a> for the default source editor shortcuts.</p>
```
@chrisdavidmills Here I couldn't copy the key mapping in because even though this is the ONLY place it was transcluded, the source editor can in theory apply to different types of source. So I chose to link back.

hamishwillee commented on files/en-us/tools/tools_toolbox/index.html:
```diff
@@ -9,10 +9,10 @@
- <li>click the wrench icon (<img alt="" src="wrench-icon.png">), which is in the main toolbar or under the Hamburger menu (<img alt="" src="hamburger.png">), then select "Toggle Tools"</li>
- <li>activate any tool hosted in the Toolbox (for example, the JavaScript Debugger or the Page Inspector)</li>
- <li>press <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>I</kbd> on Windows and Linux, or <kbd>Cmd</kbd> + <kbd>Opt</kbd> + <kbd>I</kbd> on OS X. See also <a href="/en-US/docs/Tools/Keyboard_shortcuts">keyboard shortcuts</a>.</li>
+ <li>Select "Toggle Tools" from the Web Developer menu (under "Tools" on OS X and Linux, or "Firefox" on Windows)</li>
```
@chrisdavidmills This is the beginning of page https://developer.mozilla.org/en-US/docs/Tools/Tools_Toolbox
I think it is a bit incorrect. I was referring to a wrench that no longer exists. Can you confirm that nowdays the ways to open the toolbar are:
- Keyboard keys
- Right click on any element and click inspect element
- On any platform (right?), use hamburger and do **Web Developer** - then do Toggle Tools or select specific tool.

There aren't any other ways right?

chrisdavidmills replied:
I think it would be better to list the three most common different ways, which you've basically got correct above, except that the hamburger menu option now seems to be Hamburger menu > More tools > Web developer tools.

hamishwillee replied:
Done!

hamishwillee commented on files/en-us/tools/eyedropper/index.html:
```diff
@@ -6,9 +6,7 @@
   - Tools
   - Web Development:Tools
 ---
-<div>{{ToolsSidebar}}</div><div class="geckoVersionNote">
```
Note, FF31 irrelevant now.

hamishwillee commented on files/en-us/tools/keyboard_shortcuts/index.html:
```diff
@@ -7,215 +7,31 @@
-<p>To select these, visit <code>about:config</code>, select the setting <code>devtools.editor.keymap</code>, and assign "vim" or "emacs", or "sublime" to that setting. If you do this, the selected bindings will be used for all the developer tools that use the source editor. You need to reopen the editor for the change to take effect.</p>
-
-<p>From Firefox 33 onwards, the key binding preference is exposed in the <a href="/en-US/docs/Tools/Settings#editor_preferences">Editor Preferences</a> section of the developer tools settings, and you can set it there instead of <code>about:config</code>.</p>
+<div class="notecard note">
```
@chrisdavidmills I couldn't remove this "Source Editor" section to another page because I don't think there is one. At the moment this is only linked by the Style editor section - is there another place you can edit things? I.e. is the specific or generic?

chrisdavidmills replied:
I'm really not sure

hamishwillee commented on files/en-us/tools/keyboard_shortcuts/index.html:
```diff
@@ -1074,37 +234,4 @@ <h2 id="Scratchpad">Scratchpad</h2>
-  </tr>
- </tbody>
-</table>
+<p>See <a href="/en-US/docs/Tools/Eyedropper#keyboard_shortcuts">Eyedropper > Keyboard Shortcuts</a>.</p>
```
@chrisdavidmills Section above is about a tool called "Scratchpad". I can't find any docs on that so I can't move this. Any ideas on what it is for?

chrisdavidmills replied:
This is gone; see https://developer.mozilla.org/en-US/docs/Tools/Deprecated_tools#scratchpad for details.

chrisdavidmills reviewed (changes requested):
Not much to change here, from the outset. 

I think this way of doing it is ok, although it does make me wonder if it would be better to just have a single page with all shortcuts listed on it, and then link the landing pages for each tool to their specific section on that page for the tool-specific shortcuts?

Then someone who wants all the shortcuts in one place can just visit that page anyway?

Thoughts? I've not got a particularly strong opinion about this.

hamishwillee commented:
> I think this way of doing it is ok, although it does make me wonder if it would be better to just have a single page with all shortcuts listed on it, and then link the landing pages for each tool to their specific section on that page for the tool-specific shortcuts?

Thanks @chrisdavidmills. That was my concern, as in the description:

>  I am not certain this was the best decision, mostly because now the shortcuts are scattered. I do think that that if you're using web console, page inspector etc you probably are happy enough with the current way. If you think I'd be better off duplicating or linking to the All ... doc instead let me know - relatively easy fix.

I think I will revert this. I'll push back to draft and fix up on Friday.

hamishwillee committed:
Add links to tools pages that use shortcuts

hamishwillee committed:
Remove page inspector keyboard shortcuts and fix up all links

hamishwillee committed:
Remove pagemacro from Eyedropper

hamishwillee committed:
Remove Web Console keyboard shortcuts page and fix up redirects

hamishwillee committed:
StyleEditor - remove page macro/crosslink

hamishwillee committed:
Fix up pagemacro link from Toolbox doc

hamishwillee committed:
Minor tidy to 3D view doc

hamishwillee commented on files/en-us/tools/tools_toolbox/index.html:
```diff
@@ -68,12 +68,11 @@ <h3 id="Extra_tools">Extra tools</h3>
  <li>Take a screenshot of the entire page: take a screenshot of the complete web page and saves it in your Downloads directory</li>
  <li><a href="/en-US/docs/Tools/Rulers">Toggle rulers for the page</a></li>
  <li>Measure a portion of the page: measure a part of the website by selecting areas within the page</li>
+ <li><a href="/en-US/docs/Tools/3D_View">3D view</a> (Removed in Firefox 47)</li>
```
@chrisdavidmills Should I just delete this link -  Firefox 47 was 2016? Note that the document could still exist, but you'd find it by search if you were on an old build

chrisdavidmills replied:
Yeah, let's get rid of it. I'll do that now.

hamishwillee committed:
Delete Debugger shortcuts page and redirect

hamishwillee committed:
Examine and edit CSS - fix fixable flaws

hamishwillee commented:
@peterbe Just FYI, this popped up - but the URL is not new.
![image](https://user-images.githubusercontent.com/5368500/117388265-cba66180-af2d-11eb-9d5e-2e20e84ad017.png)

hamishwillee commented:
@chrisdavidmills Ready for your re-review. 
As we discussed this inverts the changes - we now keep the single keyboard shortcut page. Where pagemacros were part of a page the inclusion was replaced with a link. Where there was a dedicated page for a specific tool's keyboard mappings I just deleted the page, and fixed up links to it from appropriate places.

chrisdavidmills committed:
remove irrelevant link

chrisdavidmills reviewed (approved):
Nice work @hamishwillee ; I think this is now ready to merge.

peterbe commented:
> @peterbe Just FYI, this popped up - but the URL is not new.
> ![image](https://user-images.githubusercontent.com/5368500/117388265-cba66180-af2d-11eb-9d5e-2e20e84ad017.png)

It gets confused because of this diff portion:
```diff
-<p><a name="mac-function-key-note">1. By default, on some Macs, the function key is remapped to use a special feature: for example, to change the screen brightness or the volume. See this </a><a href="https://support.apple.com/kb/HT3399">guide to using these keys as standard function keys</a>. To use a remapped key as a standard function key, hold the Function key down as well (so to open the Profiler, use <kbd>Shift</kbd> + <kbd>Function</kbd> + <kbd>F5</kbd>).</p>
+<p><a id="mac-function-key-note">1.</a> By default, on some Macs, the function key is remapped to use a special feature: for example, to change the screen brightness or the volume. See this <a href="https://support.apple.com/kb/HT3399">guide to using these keys as standard function keys</a>. To use a remapped key as a standard function key, hold the Function key down as well (so to open the Profiler, use <kbd>Shift</kbd> + <kbd>Function</kbd> + <kbd>F5</kbd>).</p>
```

I think the simple solution might be to change the messaging from:
```diff
-Note! This is a new URL 👀
+Note! This is maybe a new URL 👀
```

CC @escattone

escattone commented:
@peterbe 👍 I like your simple solution, but would suggest the wording `Note! This may be a new URL 👀`.

peterbe commented:
> @peterbe 👍 I like your simple solution, but would suggest the wording `Note! This may be a new URL 👀`.

https://github.com/mdn/yari/pull/3752
